packages/vitest/src/integrations/jest-mock.ts

```diff
@@ -120,7 +120,7 @@ export function spyOn<T, G extends Properties<Required<T>>>(
 export function spyOn<T, M extends Classes<Required<T>>>(
   object: T,
   method: M
-): Required<T>[M] extends new (...args: infer A) => infer R
+): Required<T>[M] extends (...args: infer A) => infer R
```

Comment:
Wouldn't this change make this type the same as `Methods` case on line 126?

From the type info it looks like you can spy on instance methods by spying on the constructor. I'm not sure if this is the case with tinyspy.

Reply:
Yes, it will. Maybe we need to change the order or something

Reply:
For context: we didn't have this overload, but added it in some issue to mock classes. This aligns with jest and works well, since classes are functions.

Reply:
I think `Classes` returns incorrect types here, since the idea here is to limit overload. `never` should never happen here in theory, so conditional types wouldn't be needed.

Reply:
I missed that part, my bad.

I'm not too familiar with TypeScript's conditional types. 
Is it possible to check if it extends a union type? Or maybe something along the lines of:
```ts
(Required<T>[M] extends (...args: infer A) => infer R) || (Required<T>[M] extends new (...args: infer A) => infer R) ? SpyInstance<A, R> : never
```

Reply:
> I missed that part, my bad.
> 
> I'm not too familiar with TypeScript's conditional types. Is it possible to check if it extends a union type? 

You need to chain them like so:

```ts
A extends B 
  ? B extends C 
    ? true
    : false
  : never
```
It's really ugly. But as I said it shouldn't be needed, since `M extends Classes<Required<T>>`, so `Required<T>[M]` should only contain constructors. `extends (...args: infer A) => infer R` is just a formality to get types of arguments and return type.

Reply:
`Classes` type is strange here. It is a dictionary of constructors for some reason. Am I missing something?

Reply:
I pushed a commit that should just merge the two overloads in a backward-compatible way.